BlinkID/example/lib/document_scanner.dart

```diff
@@ -0,0 +1,173 @@
+                          builder: (_, firstSideScanned, __) => Text(
+                                firstSideScanned ? 'Scan the back of the document.' : 'Scan the front of the document.',
+                                style: Theme.of(context)
+                                    .textTheme
+                                    .headlineLarge
+                                    ?.copyWith(color: Colors.white, shadows: _shadows),
+                              )),
+                      const SizedBox(height: 20),
+                      ValueListenableBuilder(
+                          valueListenable: _detectionStatusNotifier,
+                          builder: (_, detectionStatus, __) => Text(
+                                detectionStatus?.directions ?? '',
+                                style: Theme.of(context)
+                                    .textTheme
+                                    .headlineMedium
+                                    ?.copyWith(color: Colors.white, shadows: _shadows),
+                              )),
+                    ],
+                  )),
+              Align(
+                  alignment: Alignment.topLeft,
+                  child: IconButton(
+                      padding: const EdgeInsets.all(40),
+                      icon: const Icon(
+                        Icons.close,
+                        color: Colors.white,
+                        shadows: _shadows,
+                        size: 28,
+                      ),
+                      onPressed: () => Navigator.of(context).pop(null))),
+            ],
+          ),
+        ),
+      );
+}
+
+class _DocumentScanner extends StatelessWidget {
+  const _DocumentScanner({
+    required this.recognizerCollection,
+    required this.onResult,
+    required this.onFirstSideScanned,
+    required this.onDetectionStatusUpdate,
+    required this.useFrontCamera,
+    required this.flipFrontCamera,
+  });
+
+  final RecognizerCollection recognizerCollection;
+  final MicroblinkScannerResultCallback onResult;
+  final VoidCallback onFirstSideScanned;
+  final ValueSetter<DetectionStatus> onDetectionStatusUpdate;
+  final bool useFrontCamera;
+  final bool flipFrontCamera;
+
+  @override
+  Widget build(BuildContext context) {
+    return ColoredBox(
+      color: Colors.black,
+      child: MicroblinkScannerWidget(
+        collection: recognizerCollection,
+        settings: DocumentVerificationOverlaySettings()
+          ..useFrontCamera = useFrontCamera
+          ..flipFrontCamera = flipFrontCamera,
+        licenseKey: Theme.of(context).platform.microblinkLicenseKey,
+        onScanDone: (scanResultState) {},
+        onResult: onResult,
+        onError: print,
+        onFirstSideScanned: onFirstSideScanned,
+        onDetectionUpdate: (update) => onDetectionStatusUpdate(update.detectionStatus),
+      ),
+    );
+  }
+}
+
+extension on DetectionStatus {
+  String get directions => switch (this) {
```

Comment:
Why? This is an example app aimed at showcasing what really happens - not some user-friendly BS. I mean `.name` that enum.

Reply:
Yes. Showcasing for a real case scenario as one would use the library in their app. They wouldn't use the enum's name but rather translate the content by themselves.

Reply:
So, why didn't you delete it?

```diff
diff --git a/BlinkID/example/lib/document_scanner.dart b/BlinkID/example/lib/document_scanner.dart
index 794d791..7cc7fc9 100644
--- a/BlinkID/example/lib/document_scanner.dart
+++ b/BlinkID/example/lib/document_scanner.dart
@@ -103,7 +103,7 @@ class _DocumentScannerDialogState extends State<_DocumentScannerDialog> {
                     ValueListenableBuilder(
                       valueListenable: _detectionStatusNotifier,
                       builder: (_, detectionStatus, __) => Text(
-                        detectionStatus?.directions ?? '',
+                        detectionStatus?.name ?? '',
                         style: Theme.of(context)
                             .textTheme
                             .headlineMedium
@@ -132,19 +132,6 @@ class _DocumentScannerDialogState extends State<_DocumentScannerDialog> {
       );
 }
 
-extension on DetectionStatus {
-  String get directions => switch (this) {
-        DetectionStatus.Failed => 'No document detected.',
-        DetectionStatus.CameraAngleTooSteep => 'Adjust the camera angle.',
-        DetectionStatus.CameraTooClose => 'Move the camera further away.',
-        DetectionStatus.CameraTooFar => 'Move the camera closer.',
-        DetectionStatus.DocumentPartiallyVisible => 'Document is partially visible.',
-        DetectionStatus.DocumentTooCloseToCameraEdge => 'Move the document away from the edge.',
-        DetectionStatus.Success => 'Document detected.',
-        DetectionStatus.FallbackSuccess => 'Fallback success.',
-      };
-}
-
 extension on TargetPlatform {
   String get microblinkLicenseKey => switch (this) {
         TargetPlatform.iOS =>
```